### PR TITLE
test: add comprehensive tests for lib/tools.js

### LINKED ISSUE #294: tech-debt: lib/tools.js has zero test coverage
## Problem

`lib/tools.js` exports `checkTools()` and `assertTools()` but has no corresponding `test/tools.test.js`. The only coverage is a single `assertTools` test in `test/edge-cases.test.js`.

## Risk

- `checkTools()` return values (empty array, partial missing) are not independently verified
- `assertTools()` error message formatting for multiple missing tools is untested
- The injectable `_exec` parameter contract is not validated

## Suggested Fix

Create `test/tools.test.js` covering:
1. All tools present → returns empty array
2. One tool missing → returns that tool name
3. Multiple tools missing → returns all missing
4. `assertTools` throws `RallyError` with `EXIT_CONFIG` and correct message
5. `assertTools` succeeds when all tools present

**Priority:** Medium — this is a small module but guards the entire CLI bootstrap path.

### PR DESCRIPTION
**Pull request by jsturtevant**
Closes #294

## Summary
Adds `test/tools.test.js` with 12 tests covering all exported functions from `lib/tools.js`:

### `checkTools`
- Returns empty array when all tools are present
- Returns all three tools (`git`, `gh`, `npx`) when none are present
- Returns only the missing tools when some are absent
- Detects multiple missing tools
- Verifies exec is called with `--version` and `{ stdio: 'pipe' }`
- Works with default opts (no arguments)

### `assertTools`
- Does not throw when all tools are present
- Throws `RallyError` when tools are missing
- Thrown error has `EXIT_CONFIG` exit code
- Error message lists all missing tools
- Error message lists only the missing tool
- Does not throw when all present

All tests use the `_exec` dependency injection seam already built into the module.

### CODE REVIEW

File: test/tools.test.js [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    } catch (err) {
+      assert.ok(err.message.includes('npx'));
+      assert.ok(!err.message.includes('git,'));
+    }
+  });
+
+  it('does not throw when a single tool is present among all', () => {
+    // All pass → no throw
+    assert.doesNotThrow(() => assertTools({ _exec: execAllPass }));
+  });
```
[the remark this answers is not in the document]
Comment (jsturtevant) [reply test/tools.test.js]: Good catch — removed the redundant test. The all-present test at line 65 already covers this.
